src/shared/nt-core/MainPage.react.js

```diff
@@ -26,7 +26,7 @@ const enhance = compose(
   graphql(queryProduct, {
     options: ({ match }) => ({ variables: { productId: match.params.productId } })
   }),
-  handleError((props) => (!props.data.product)),
+  handleError(({ data }) => (!data.product && !data.loading)),
```

Comment:
https://github.com/TheInteract/Apollo/pull/6#discussion_r111659052
I have already created a new higher order component (handleError) to handle any error which is occurred before doing a withLoader middleware.